src/functions-and-arrays.js

```diff
@@ -1,41 +1,151 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(number1,number2) {
+    if(number1>number2){
+      return number1
+    }
+    else if (number2>number1){
+      return number2
+    }
+    else {
+      return number1
+    };
+};
+maxOfTwoNumbers(3,5);
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
-
+function findLongestWord(wordsArray) {
+  if( wordsArray === null || wordsArray === undefined || wordsArray.length === 0){
+    return null;
+  }
+  else{
+  let max = 0;
+  let wordsLength = [];
+  for(i=0;i<wordsArray.length;i++){
+      wordsLength.push(wordsArray[i].length)
+      if(wordsArray[i].length>max){
+          max = wordsArray[i].length
+      }
+      else if(wordsArray[i].length<=max){
+          continue;
+      };
+  };
+  return wordsArray[wordsLength.indexOf(max)];
+ }
+}
+findLongestWord(words)
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbersArray) {
+  if( numbersArray === null || numbersArray === undefined || numbersArray.length === 0){
+    return 0;
+  }
+  else{
+    let count=0;
+    for (i=0;i<numbersArray.length;i++){
+      count = count + numbersArray[i]
+    };
+    return count
+  }
+}
+
+sumNumbers(numbers)
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+const objectTest ={age:32}
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+// should return: 57
+
+function sum(array) {
+  if( array === null || array === undefined || array.length === 0){
+    return 0;
+  }
+  
+  else{
+    let count = 0
+    for(i=0;i<array.length;i++){
+      switch(typeof array[i]) {
+        case "number": count += array[i] ;break;
+        case "string": count += array[i].length ;break;
+        case "boolean": count += Number(array[i]);break;
+        case "array": continue;
+        case "object": continue;
+      }
+    }
+    console.log(count)
+    return count
+  }
+}
+sum (mixedArr)
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(avgArray) {
+  if(avgArray === null || avgArray === undefined || avgArray.length === 0){
+    return null;
+  }
+    else{
+  let count=0;
```

Comment:
Since you previously created a function to sum numbers, you should reuse it here instead of repeating the whole logic to sum the numbers.
```
let count = sumNumbers(avgArray);
return count / avgArray.length
```